pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/LLCSegmentCommitUpload.java

```diff
@@ -0,0 +1,128 @@
+  public Representation post(Representation entity) {
+    SegmentCompletionProtocol.Request.Params params = SegmentCompletionUtils.extractParams(getReference());
+    if (params == null) {
+      return new StringRepresentation(SegmentCompletionProtocol.RESP_FAILED.toJsonString());
+    }
+
+    LOGGER.info(params.toString());
+
+    boolean success = uploadSegment(params.getInstanceId(), params.getSegmentName(), params.getOffset());
+    if (success) {
+      LOGGER.info("Uploaded segment successfully");
+      return new StringRepresentation(SegmentCompletionProtocol.ControllerResponseStatus.UPLOAD_SUCCESS.toString());
+    } else {
+      LOGGER.info("Failed to upload segment");
+      return new StringRepresentation(SegmentCompletionProtocol.ControllerResponseStatus.FAILED.toString());
+    }
+  }
+
+  boolean uploadSegment(final String instanceId, final String segmentNameStr, final long offset) {
+    // 1/ Create a factory for disk-based file items
+    final DiskFileItemFactory factory = new DiskFileItemFactory();
+
+    // 2/ Create a new file upload handler based on the Restlet
+    // FileUpload extension that will parse Restlet requests and
+    // generates FileItems.
+    final RestletFileUpload upload = new RestletFileUpload(factory);
+    final List<FileItem> items;
+
+    try {
+      // The following statement blocks until the entire segment is read into memory/disk.
+      items = upload.parseRequest(getRequest());
+
+      File dataFile = null;
+
+      // TODO: refactor this part into a util method (almost duplicate code in PinotSegmentUploadRestletResource and
+      // PinotSchemaRestletResource)
+      for (FileItem fileItem : items) {
+        String fieldName = fileItem.getFieldName();
+        if (dataFile == null) {
+          if (fieldName != null && fieldName.equals(segmentNameStr)) {
+            dataFile = new File(tempDir, fieldName);
+            fileItem.write(dataFile);
+          } else {
+            LOGGER.warn("Invalid field name: {}", fieldName);
+          }
+        } else {
+          LOGGER.warn("Got extra file item while uploading LLC segments: {}", fieldName);
+        }
+
+        // Remove the temp file
+        // When the file is copied to instead of renamed to the new file, the temp file might be left in the dir
+        fileItem.delete();
+      }
+
+      if (dataFile == null) {
+        LOGGER.error("Segment not included in request. Instance {}, segment {}", instanceId, segmentNameStr);
+        return false;
+      }
+      // We will not check for quota here. Instead, committed segments will count towards the quota of a
+      // table
+      LLCSegmentName segmentName = new LLCSegmentName(segmentNameStr);
+
+      final String rawTableName = segmentName.getTableName();
+      final File tableDir = new File(baseDataDir, rawTableName);
+      final File segmentFile = new File(tableDir, SegmentCompletionUtils.generateSegmentFileName(segmentNameStr, offset, instanceId));
```

Comment:
pull out the code to genearte file name as:
generatedFileName  = SCU.generateFileName(..)